core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -386,8 +386,8 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
         }
 
         @Override
-        public Query fuzzyQuery(String value, Fuzziness fuzziness, int prefixLength, int maxExpansions, boolean transpositions) {
-            long iValue = dateMathParser().parse(value, now());
+        public Query fuzzyQuery(Object value, Fuzziness fuzziness, int prefixLength, int maxExpansions, boolean transpositions) {
+            long iValue = dateMathParser().parse(value.toString(), now());
```

Comment:
Can you use parseValue(value) instead? It should behave as similarly to termQuery() as possible.